lib/spack/spack/test/conftest.py

```diff
@@ -12,9 +12,21 @@
 import os.path
 import re
 import shutil
+import sys
 import tempfile
 import xml.etree.ElementTree
 
+if sys.version_info >= (2, 7):
```

Comment:
@eschnett according to https://github.com/spack/spack/pull/24471/checks?check_run_id=2887741511 this is also not supported for Python 2.7 either (at least on our CI systems). I'm not sure why but that didn't get caught in the CI for this PR, but I think this statement will have to skip import of `dateutil` for all Python 2.x. Do you have a minute to update this? I'm trying to get time ASAP but won't be able to for another hour or so.

Reply:
I've made an attempt at this at https://github.com/spack/spack/pull/24473

Reply:
Thanks!

Reply:
I still see errors like the one above (e.g. at https://github.com/spack/spack/pull/24388/checks?check_run_id=2890513500). I don't think this is related to Python 2.7, because it's not the CVS-related code that is failing.

I think what is going on is the following: As part of the test suite, there is an analysis stage (related to `external/_pytest`) which analyzes the Python code. This analysis stage comes across the `import dateutil` statement and cannot handle it.

I don't know what `external/_pytest` does, or what analysis stage that is. But maybe it would suffice to tell this stage that `dateutil` is now a fine package to be imported, and to not worry about it.

Reply:
I am surprised: at least one PR submitted since #24473 has passed CI (https://github.com/spack/spack/pull/24481) so I'm wondering if there is an issue where PRs that failed due to this are difficult to correct (for the PR you linked, it hit this problem, and you pushed additional commits after it was corrected, but I see the CI fails in both cases).

Reply:
Posting a detail that I noticed on a PR of mine #24398 that is being hit by the same issue. When the error happens it seems Spack picks Python 3.8 as the interpreter, even if it is in the Python 2.7 CI job. See [here](https://github.com/spack/spack/pull/24398/checks?check_run_id=2892195010) for the log of the failure.

Reply:
I think this uncovered some kind of subtle bug where package only PRs are running on Python 3.8 when in the Python 2.7 CI job